### PR TITLE
Force email address (username) to lowercase in all registrations/logins

### PR DESCRIPTION
**Pull request by irlaec**

### CODE REVIEW

**Comment by irlaec** [comment]
[Login: make email addresses not case sensitive](https://trello.com/c/8M1meUEQ/76-login-make-email-addresses-not-case-sensitive)

**Comment by irlaec** [comment]
Fixes #37

File: octoprint_mrbeam/static/js/loginscreen_viewmodel.js [hunk 1]
```diff
@@ -2,10 +2,23 @@
  * Created by andy on 03/03/2017.
  */
 $(function() {
+    $('#settings-usersDialogAddUserName').attr('data-bind','value: $root.users.editorUsername, valueUpdate: \'afterkeydown\'');
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/loginscreen_viewmodel.js:6]:
```suggestion
    // MR_BEAM_OCTOPRINT_PRIVATE_API_ACCESS
    $('#settings-usersDialogAddUserName').attr('data-bind','value: $root.users.editorUsername, valueUpdate: \'afterkeydown\'');
```

File: octoprint_mrbeam/static/js/loginscreen_viewmodel.js [hunk 2]
```diff
@@ -2,10 +2,23 @@
...
+
     function LoginScreenViewModel(parameters) {
         var self = this;
 
+        window.ko.extenders.lowercase = function(target, option) {
+            target.subscribe(function(newValue) {
+                if(newValue !== undefined) {
+                    target(newValue.toLowerCase());
+                    console.log(newValue)
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/loginscreen_viewmodel.js:15]: please remove or make the log output more explanatory

File: octoprint_mrbeam/static/js/wizard_acl.js [hunk 3]
```diff
@@ -56,6 +56,7 @@ $(function() {
             // needs to be scrollable on touch devices
             $('#wizard_dialog .modal-body').addClass('scrollable');
             $('#wizard_dialog div.modal-footer div.text-center').hide();
+            self.username = ko.observable(undefined).extend({lowercase: true});
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/wizard_acl.js:59]: Does undefined work? I would suggest an empty string as default val. No problem if it works though.

**Review by andywerner — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/static/js/wizard_acl.js (hunk 3 above)
Comment (irlaec) [reply octoprint_mrbeam/static/js/wizard_acl.js:59]: Works :)

File: octoprint_mrbeam/static/js/loginscreen_viewmodel.js (hunk 2 above)
Comment (irlaec) [reply octoprint_mrbeam/static/js/loginscreen_viewmodel.js:15]: Deleted

**Review by andywerner — APPROVED** [review]

**Pull request merged**